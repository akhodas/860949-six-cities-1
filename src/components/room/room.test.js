import React from 'react';
import renderer from 'react-test-renderer';
import {MemoryRouter} from 'react-router';

import Room from './room.jsx';

describe(`Room`, () => {
  it(`correctly rendered with isLoadData=true`, () => {
    const tree = renderer
      .create(<MemoryRouter> <Room
        room={{
          id: 0,
          city: {
            name: `Minsk`,
            location: {
              latitude: 0,
              longitude: 0,
              zoom: 5,
            },
          },
          previewImage: `path`,
          images: [`path1`, `path2`],
          title: `title`,
          isFavorite: true,
          isPremium: true,
          rating: 0,
          type: `type`,
          bedrooms: 0,
          maxAdults: 0,
          price: 0,
          goods: [`goods1`, `goods2`],
          host: {
            id: 1,
            isPro: true,
            name: `name`,
            avatarUrl: `pathAvatar`
          },
          description: `description`,
          location: {
            latitude: 1,
            longitude: 1,
            zoom: 8
          }
        }}
        images={[`path1`, `path2`]}
        emailUser={`myEmail@mail.com`}
        isAuthorizationStatus={true}
        controlAuthorization={jest.fn()}
        isLoadData={true}
        comments={[]}
      /> </MemoryRouter>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`correctly rendered with isLoadData=false`, () => {
    const tree = renderer
      .create(<MemoryRouter> <Room
        room={{
          id: 0,
          city: {
            name: `Minsk`,
            location: {
              latitude: 0,
              longitude: 0,
              zoom: 5,
            },
          },
          previewImage: `path`,
          images: [`path1`, `path2`],
          title: `title`,
          isFavorite: true,
          isPremium: true,
          rating: 0,
          type: `type`,
          bedrooms: 0,
          maxAdults: 0,
          price: 0,
          goods: [`goods1`, `goods2`],
          host: {
            id: 1,
            isPro: true,
            name: `name`,
            avatarUrl: `pathAvatar`
          },
          description: `description`,
          location: {
            latitude: 1,
            longitude: 1,
            zoom: 8
          }
        }}
        images={[`path1`, `path2`]}
        emailUser={`myEmail@mail.com`}
        isAuthorizationStatus={true}
        controlAuthorization={jest.fn()}
        isLoadData={true}
        comments={[]}
      /> </MemoryRouter>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
