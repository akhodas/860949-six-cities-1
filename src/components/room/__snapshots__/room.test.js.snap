// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Room correctly rendered with flagDataIsLoading=false 1`] = `
Array [
  " ",
  <div
    style={
      Object {
        "display": "none",
      }
    }
  >
    <svg
      xmlns="http://www.w3.org/2000/svg"
    >
      <symbol
        id="icon-arrow-select"
        viewBox="0 0 7 4"
      >
        <path
          clipRule="evenodd"
          d="M0 0l3.5 2.813L7 0v1.084L3.5 4 0 1.084V0z"
          fillRule="evenodd"
        />
      </symbol>
      <symbol
        id="icon-bookmark"
        viewBox="0 0 17 18"
      >
        <path
          d="M3.993 2.185l.017-.092V2c0-.554.449-1 .99-1h10c.522 0 .957.41.997.923l-2.736 14.59-4.814-2.407-.39-.195-.408.153L1.31 16.44 3.993 2.185z"
        />
      </symbol>
      <symbol
        id="icon-star"
        viewBox="0 0 13 12"
      >
        <path
          clipRule="evenodd"
          d="M6.5 9.644L10.517 12 9.451 7.56 13 4.573l-4.674-.386L6.5 0 4.673 4.187 0 4.573 3.549 7.56 2.483 12 6.5 9.644z"
          fillRule="evenodd"
        />
      </symbol>
    </svg>
  </div>,
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="../img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/login"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__login"
                >
                  Sign in
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>,
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
    >
      <div
        className="property__gallery-container container"
      >
        <div
          className="property__gallery"
        >
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="path1"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="path2"
            />
          </div>
        </div>
      </div>
      <div
        className="property__container container"
      >
        <div
          className="property__wrapper"
        >
          <div
            className="property__mark"
          >
            <span>
              Premium
            </span>
          </div>
          <div
            className="property__name-wrapper"
          >
            <h1
              className="property__name"
            >
              title
            </h1>
            <button
              className="property__bookmark-button property__bookmark-button--active button"
              onClick={[Function]}
              type="button"
            >
              <svg
                className="property__bookmark-icon"
                height="33"
                width="31"
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="property__rating rating"
          >
            <div
              className="property__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "0%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
            <span
              className="property__rating-value rating__value"
            >
              0
            </span>
          </div>
          <ul
            className="property__features"
          >
            <li
              className="property__feature property__feature--entire"
            >
              type
            </li>
            <li
              className="property__feature property__feature--bedoffers"
            >
              0
               Bedrooms
            </li>
            <li
              className="property__feature property__feature--adults"
            >
              Max 
              0
               adults
            </li>
          </ul>
          <div
            className="property__price"
          >
            <b
              className="property__price-value"
            >
              €
              0
            </b>
            <span
              className="property__price-text"
            >
               night
            </span>
          </div>
          <div
            className="property__inside"
          >
            <h2
              className="property__inside-title"
            >
              What's inside
            </h2>
            <ul
              className="property__inside-list"
            >
              <li
                className="property__inside-item"
              >
                goods1
              </li>
              <li
                className="property__inside-item"
              >
                goods2
              </li>
            </ul>
          </div>
          <div
            className="property__host"
          >
            <h2
              className="property__host-title"
            >
              Meet the host
            </h2>
            <div
              className="property__host-user user"
            >
              <div
                className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
              >
                <img
                  alt="Host avatar"
                  className="property__avatar user__avatar"
                  height="74"
                  src="pathAvatar"
                  width="74"
                />
              </div>
              <span
                className="property__user-name"
              >
                name
              </span>
              <span
                className="property__user-status"
              >
                Pro
              </span>
            </div>
            <div
              className="property__description"
            >
              <p
                className="property__text"
              >
                description
              </p>
            </div>
          </div>
          <section
            className="property__reviews reviews"
          >
            <h2
              className="reviews__title"
            >
              Reviews ·
              <span
                className="reviews__amount"
              >
                0
              </span>
            </h2>
            <ul
              className="reviews__list"
            />
          </section>
        </div>
      </div>
      <div
        style={
          Object {
            "margin": "auto",
            "width": "990px",
          }
        }
      >
        <section
          className="property__map map"
        >
          <div
            id="map"
            style={
              Object {
                "height": "100%",
              }
            }
          />
        </section>
      </div>
    </section>
    <div
      className="container"
    >
      <section
        className="near-places places"
      >
        <h2
          className="near-places__title"
        >
          Other places in the neighbourhood
        </h2>
        <div
          className="near-places__list places__list"
        >
          <article
            className="near-places__card place-card"
          >
            <div
              className="place-card__mark"
            >
              <span>
                Premium
              </span>
            </div>
            <div
              className="near-places__image-wrapper place-card__image-wrapper"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                <img
                  alt="Place image"
                  className="place-card__image"
                  height="200"
                  src="path"
                  width="260"
                />
              </a>
            </div>
            <div
              className="place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    0
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button place-card__bookmark-button--active button"
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "0%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  title
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                type
              </p>
            </div>
          </article>
        </div>
      </section>
    </div>
  </main>,
  " ",
]
`;

exports[`Room correctly rendered with flagDataIsLoading=true 1`] = `
Array [
  " ",
  <div
    style={
      Object {
        "display": "none",
      }
    }
  >
    <svg
      xmlns="http://www.w3.org/2000/svg"
    >
      <symbol
        id="icon-arrow-select"
        viewBox="0 0 7 4"
      >
        <path
          clipRule="evenodd"
          d="M0 0l3.5 2.813L7 0v1.084L3.5 4 0 1.084V0z"
          fillRule="evenodd"
        />
      </symbol>
      <symbol
        id="icon-bookmark"
        viewBox="0 0 17 18"
      >
        <path
          d="M3.993 2.185l.017-.092V2c0-.554.449-1 .99-1h10c.522 0 .957.41.997.923l-2.736 14.59-4.814-2.407-.39-.195-.408.153L1.31 16.44 3.993 2.185z"
        />
      </symbol>
      <symbol
        id="icon-star"
        viewBox="0 0 13 12"
      >
        <path
          clipRule="evenodd"
          d="M6.5 9.644L10.517 12 9.451 7.56 13 4.573l-4.674-.386L6.5 0 4.673 4.187 0 4.573 3.549 7.56 2.483 12 6.5 9.644z"
          fillRule="evenodd"
        />
      </symbol>
    </svg>
  </div>,
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="../img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/login"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__login"
                >
                  Sign in
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>,
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
    >
      <div
        className="property__gallery-container container"
      >
        <div
          className="property__gallery"
        >
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="path1"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="path2"
            />
          </div>
        </div>
      </div>
      <div
        className="property__container container"
      >
        <div
          className="property__wrapper"
        >
          <div
            className="property__mark"
          >
            <span>
              Premium
            </span>
          </div>
          <div
            className="property__name-wrapper"
          >
            <h1
              className="property__name"
            >
              title
            </h1>
            <button
              className="property__bookmark-button property__bookmark-button--active button"
              onClick={[Function]}
              type="button"
            >
              <svg
                className="property__bookmark-icon"
                height="33"
                width="31"
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="property__rating rating"
          >
            <div
              className="property__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "0%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
            <span
              className="property__rating-value rating__value"
            >
              0
            </span>
          </div>
          <ul
            className="property__features"
          >
            <li
              className="property__feature property__feature--entire"
            >
              type
            </li>
            <li
              className="property__feature property__feature--bedoffers"
            >
              0
               Bedrooms
            </li>
            <li
              className="property__feature property__feature--adults"
            >
              Max 
              0
               adults
            </li>
          </ul>
          <div
            className="property__price"
          >
            <b
              className="property__price-value"
            >
              €
              0
            </b>
            <span
              className="property__price-text"
            >
               night
            </span>
          </div>
          <div
            className="property__inside"
          >
            <h2
              className="property__inside-title"
            >
              What's inside
            </h2>
            <ul
              className="property__inside-list"
            >
              <li
                className="property__inside-item"
              >
                goods1
              </li>
              <li
                className="property__inside-item"
              >
                goods2
              </li>
            </ul>
          </div>
          <div
            className="property__host"
          >
            <h2
              className="property__host-title"
            >
              Meet the host
            </h2>
            <div
              className="property__host-user user"
            >
              <div
                className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
              >
                <img
                  alt="Host avatar"
                  className="property__avatar user__avatar"
                  height="74"
                  src="pathAvatar"
                  width="74"
                />
              </div>
              <span
                className="property__user-name"
              >
                name
              </span>
              <span
                className="property__user-status"
              >
                Pro
              </span>
            </div>
            <div
              className="property__description"
            >
              <p
                className="property__text"
              >
                description
              </p>
            </div>
          </div>
          <section
            className="property__reviews reviews"
          >
            <h2
              className="reviews__title"
            >
              Reviews ·
              <span
                className="reviews__amount"
              >
                0
              </span>
            </h2>
            <ul
              className="reviews__list"
            />
          </section>
        </div>
      </div>
      <div
        style={
          Object {
            "margin": "auto",
            "width": "990px",
          }
        }
      >
        <section
          className="property__map map"
        >
          <div
            id="map"
            style={
              Object {
                "height": "100%",
              }
            }
          />
        </section>
      </div>
    </section>
    <div
      className="container"
    >
      <section
        className="near-places places"
      >
        <h2
          className="near-places__title"
        >
          Other places in the neighbourhood
        </h2>
        <div
          className="near-places__list places__list"
        >
          <article
            className="near-places__card place-card"
          >
            <div
              className="place-card__mark"
            >
              <span>
                Premium
              </span>
            </div>
            <div
              className="near-places__image-wrapper place-card__image-wrapper"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                <img
                  alt="Place image"
                  className="place-card__image"
                  height="200"
                  src="path"
                  width="260"
                />
              </a>
            </div>
            <div
              className="place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    0
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button place-card__bookmark-button--active button"
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "0%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  title
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                type
              </p>
            </div>
          </article>
        </div>
      </section>
    </div>
  </main>,
  " ",
]
`;
